native/src/main/java/io/ballerina/stdlib/serdes/MessageSerializer.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.serdes;
+
+import com.google.protobuf.DynamicMessage;
+import com.google.protobuf.DynamicMessage.Builder;
+import io.ballerina.runtime.api.values.BArray;
+import io.ballerina.runtime.api.values.BDecimal;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.runtime.api.values.BTable;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+import static com.google.protobuf.Descriptors.Descriptor;
+import static com.google.protobuf.Descriptors.FieldDescriptor;
+import static io.ballerina.stdlib.serdes.Constants.NIL;
+import static io.ballerina.stdlib.serdes.Constants.PRECISION;
+import static io.ballerina.stdlib.serdes.Constants.SCALE;
+import static io.ballerina.stdlib.serdes.Constants.UNSUPPORTED_DATA_TYPE;
+import static io.ballerina.stdlib.serdes.Constants.VALUE;
+import static io.ballerina.stdlib.serdes.Utils.SERDES_ERROR;
+import static io.ballerina.stdlib.serdes.Utils.createSerdesError;
+
+/**
+ * {@link MessageSerializer} provides generic functions for concrete message serializers.
+ */
+public abstract class MessageSerializer {
+    private final Builder dynamicMessageBuilder;
+    private final Object anydata;
```

Comment:
What does this field store?

Reply:
Stores the value of the ballerina structure types.

Reply:
Shall we update the field name to a more explanatory name then? `anydata` has a specific value in the Ballerina context and this isn't exactly used in that terms.

Reply:
+1, will change to `ballerinaStructureTypeValue`

Reply:
Updated